core/services/relay/evm/types/types.go

```diff
@@ -56,6 +56,14 @@ type chainReaderDefinitionFields struct {
 
 	// EventInputFields allows you to choose which indexed fields are expected from the input
 	EventInputFields []string `json:"eventInputFields,omitempty"`
+
+	// QueryingKeys specifies for which topics to create key querying pairs. Key is topicIndex, value is bool.
+	// Implementation should create <key>-<subdata> pairs for each specified topic.
+	// If we use EVM as an eg. it is <genericEventName>-<topicName> mapping (this is not restricted to just indexed topics).
+	QueryingKeys map[int]bool `json:"eventQueryingMappings,omitempty"`
```

Comment:
nit/ Would it make sense to say just `QueryKeys`?
```suggestion
	// QueryKeys specifies for which topics to create key query pairs. Key is topicIndex, value is bool.
	// Implementation should create <key>-<subdata> pairs for each specified topic.
	// If we use EVM as an eg. it is <genericEventName>-<topicName> mapping (this is not restricted to just indexed topics).
	QueryKeys map[int]bool `json:"eventQueryMappings,omitempty"`
```
Also, can/should the json name be more aligned?

Reply:
Yeah for sure, the PR is very rough right now. I'm still focused on the overall concept, haven't gotten around to fixing stuff like this